presentation/frontend/src/presentation-frontend/favorite-properties/FavoritePropertiesStorage.ts

```diff
@@ -49,6 +49,31 @@ export interface IFavoritePropertiesStorage {
   savePropertiesOrder(orderInfos: FavoritePropertiesOrderInfo[], projectId: string | undefined, imodelId: string): Promise<void>;
 }
 
+/**
+ * Available implementations of [[IFavoritePropertiesStorage]].
+ * @public
+ */
+export enum DefaultFavoritePropertiesStorageTypes {
+  /** A no-op storage that doesn't store or return anything. Used for cases when favorite properties aren't used by the application. */
+  Noop,
+  /** A storage that stores favorite properties information in a browser local storage. */
+  BrowserLocalStorage,
+  /** A storage that stores favorite properties in a user settings service (see [[IModelApp.settings]]). */
+  UserSettingsServiceStorage,

```

Comment:
We are not going to be able to use PRDSS anymore and it will need to be configured by whoever is calling `iModelApp.startup` how they handle user preferences such as favorite properties.

I'm working on a PR to remove the settings service now. However, as long as there is a default fallback (even if a no-op) then that should be fine.

Reply:
I haven't read through this PR yet. Going through Keith's now, then I'll hop over to this one.

Reply:
After reviewing the PR, I think this enum can stay and use the `IModelApp.settings` API but should avoid any reference to a Bentley service as the default for it.

The plan is to keep the `IModelApp.settings` API (although may rename), but not provide a default implemenation anywhere with iTwin.js Core so that will fully be on someone else to provide. If they do provide one, then they will need to both supply it to the `IModelApp.startup` and then select this Enum in `Presentation.initialize` methods.

Reply:
@grigasp, @aruniverse FYI, here's a draft PR #2291 with the changes I'm intending to make.  I'll have to merge with master before I finish the changes needed to presentation.